wfe2/verify.go

```diff
@@ -473,24 +473,18 @@ func (wfe *WebFrontEndImpl) validJWSForKey(
 	return payload, nil
 }
 
-// validPOSTForAccount checks that a given POST request has a valid JWS
-// verified with the public key associated to a known account,
-// specified by the JWS key ID. If the request is valid (e.g. the JWS is well
-// formed, verifies with the JWK stored for the specified key ID, specifies the
-// correct URL, and has a valid nonce) then `validPOSTForAccount` returns the
-// validated JWS body, the parsed JSONWebSignature, and a pointer to the JWK's
-// associated account. If any of these conditions are not met or an error
-// occurs only a problem is returned.
-func (wfe *WebFrontEndImpl) validPOSTForAccount(
+// validJWSForAccount checks that a given JWS is valid and verifies with the
+// public key associated to a known account specified by the JWS Key ID. If the
+// JWS is valid (e.g. the JWS is well formed, verifies with the JWK stored for the
+// specified key ID, specifies the correct URL, and has a valid nonce) then
+// `validJWSForAccount` returns the validated JWS body, the parsed
+// JSONWebSignature, and a pointer to the JWK's associated account. If any of
+// these conditions are not met or an error occurs only a problem is returned.
+func (wfe *WebFrontEndImpl) validJWSForAccount(
+	jws *jose.JSONWebSignature,
 	request *http.Request,
 	ctx context.Context,
 	logEvent *requestEvent) ([]byte, *jose.JSONWebSignature, *core.Registration, *probs.ProblemDetails) {
```

Comment:
Keeping this return value makes it slightly easier to use from `validPOSTForAccount` since it needs to return the JWS and this way I can `return validJWSForAccount(jws, request, ctx, logEvent)`.